protocol/contracts/governance/Templar.sol

```diff
@@ -0,0 +1,89 @@
+//SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+/**
+ * @title an NFT representing membership in the temple community.
+ *
+ * It exists as on chain proof that an account owns a discord id.
+ */
+contract Templar is ERC721, AccessControl {
+
+    bytes32 public constant OWNER = keccak256("OWNER");
+    bytes32 public constant CAN_ASSIGN = keccak256("CAN_ASSIGN");
+
+    string public baseUri = "https://discordapp.com/users/";
+
+     // Mapping from discordId ID to max temple role
+    mapping(uint256 => string) public templeRole;
```

Comment:
I considered that, however I decide to go with strings so that the contract will continue to work when roles are changed 

This is will happen as "master" is removed and "higher initiate" is added. I feel like roles are likely to change further as gov is refined.

Reply:
Strings for now, 100% roles will change like crazy :p

Reply:
If normalising to optimise gas/space was indeed critical (which I don't think it is given the number of discordIds we'll need):
Could also have a dynamic map linking an `roleIndex->roleString`, so then the value of templeRole is also a uint256.

Reply:
I'd be open to that if we need it, but let's keep things simple for now.